src/app/client/src/app/modules/dashboard/components/data-chart/data-chart.component.ts

```diff
@@ -398,13 +399,22 @@ export class DataChartComponent implements OnInit, OnDestroy {
   @Input()
   set globalFilter(val: any) {
     if(val){
-      val.chartData['selectedFilters'] = {};
-      this.chartData = val.chartData;
-      this.chartData['selectedFilters'] = { };
-      this.cdr.detectChanges();
-      this.getDataSetValue(val.chartData);
-      this.calculateBigNumber(val.chartData);
-      this.resetForm();
+      let updatedChartData = val.chartData.filter(chart=>{
```

Comment:
Please add a nullish check here . Whatif the val or chartData is undefined or null. code will break.